src/processor.cpp

```diff
@@ -0,0 +1,164 @@
+// This file is a part of Julia. License is MIT: https://julialang.org/license
+
+// Processor feature detection
+
+#include "julia.h"
+#include "julia_internal.h"
+
+extern "C" {
```

Comment:
There'll be string manipulation functions here for parsing the command line args, parsing `/proc/cpuinfo`, generating LLVM feature string and I don't really want to do that in C (especially not for the output part). Basically I want to put the processor detection logic together instead of in 3+ files. This include the codegen initialization code at https://github.com/JuliaLang/julia/blob/3d5685a851af1068e9c8018d588f762a6188582d/src/codegen.cpp#L7089